File: app/src/main/java/com/imangazalievm/bubbble/domain/interactors/UserProfileInteractor.java
package com.imangazalievm.bubbble.domain.interactors;


import com.imangazalievm.bubbble.domain.models.User;
import com.imangazalievm.bubbble.domain.repository.UsersRepository;

import javax.inject.Inject;

import io.reactivex.Single;

public class UserProfileInteractor {

    private UsersRepository usersRepository;

    @Inject
    public UserProfileInteractor(UsersRepository usersRepository) {
        this.usersRepository = usersRepository;
    }

    public Single<User> getUser(long userId) {
        return usersRepository.getUser(userId);
    }

}

File: app/src/main/java/com/bubbble/domain/userprofile/UserProfileInteractor.kt
package com.bubbble.domain.userprofile

import com.bubbble.data.users.UsersRepository
import com.bubbble.core.models.user.User
import javax.inject.Inject

class UserProfileInteractor @Inject constructor(
    private val usersRepository: UsersRepository
) {

    suspend fun getUser(userId: Long): User {
        return usersRepository.getUser(userId)
    }

}